tests/serialization_tests/tests/signing_ser.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+// Doesn't run these unless feature specified
+#![cfg(feature = "submodule_tests")]
+#![allow(dead_code, non_snake_case)]
+
+use bls_signatures::{PrivateKey, Serialize};
+use crypto::{signature, Signature};
+use encoding::Cbor;
+use forest_message::{unsigned_message, UnsignedMessage};
+use serde::Deserialize;
+use std::fs::File;
+use std::io::prelude::*;
+
+#[derive(Deserialize)]
+struct TestVec {
+    #[serde(with = "unsigned_message::json")]
+    Unsigned: UnsignedMessage,
+    Cid: String,
+    CidHexBytes: String,
+    PrivateKey: String,
+    #[serde(with = "signature::json")]
+    Signature: Signature,
+}
+
+#[test]
+fn signing_test() {
+    let mut file = File::open("../serialization-vectors/message_signing.json").unwrap();
+    let mut string = String::new();
+    file.read_to_string(&mut string).unwrap();
+
+    let vectors: Vec<TestVec> =
+        serde_json::from_str(&string).expect("Test vector deserialization failed");
+    for test_vec in vectors {
+        let test = base64::decode(test_vec.PrivateKey).unwrap();
+        // TODO set up a private key based on sig type
+        let priv_key = PrivateKey::from_bytes(&test).unwrap();
+        let cid = test_vec.Unsigned.cid().unwrap();
+        let sig = priv_key.sign(cid.to_bytes().as_slice());
+        assert_eq!(sig.as_bytes().as_slice(), test_vec.Signature.bytes());
```

Comment:
One thing that is missing is the test for the signed message Cid. The intention of the test is that the signed message cid matches this. The test is setup pretty bad because there are only bls signed messages and the signed message cid is just the cid of the unsigned message, but there should still be the check for when secp signed message vectors are added